csm/conf/setup.py

```diff
@@ -870,3 +870,34 @@ def refresh_config(self, args):
             Log.logger.info(f"Resolved and acknowledged all the faulty node : {node_id} alerts")
         except Exception as e:
             raise CsmSetupError("csm_setup refresh_config failed. Error: %s" %e)
+
+    def post_update(self, args):
+        # Execute 'csm_setup post_update' mannually once system is updated using SW update.
+        try:
+            Log.info(f"Triggering csm_setup post_update: {args}")
+            if self._is_user_exist():
+                Log.debug(f"Deleting user {self._user}")
+                Setup._run_cmd(f"userdel -r {self._user}")
+                Setup.Config.delete()
+                Log.debug("Applying salt state post update")
+                SaltWrappers.get_salt_call("state.apply", "components.system.config.pillar_encrypt")
+                Log.debug("Execute csm_setup cmds")
+                Setup._run_cmd("csm_setup post_install")
```

Comment:
What if either of the 3 commands fail?

Reply:
All the three commands raises the error on failure. Error logged with tracebacks